test/tracer.js

```diff
@@ -0,0 +1,230 @@
+            assert.isOk(bufferEqual(span.context().parentId, parentId));
+            assert.equal(span.context().flags, constants.SAMPLED_MASK);
+            assert.equal(span._start, startTime);
+        }
+
+        assertByStartSpanParameters(childOfParams);
+        assertByStartSpanParameters(referenceParams);
+    });
+
+    it ('inject plain text headers into carrier, and extract span context with the same value', () => {
+        let keyOne = 'keyOne';
+        let keyTwo = 'keyTwo';
+        let baggage = {
+            keyOne: 'leela',
+            keyTwo: 'bender'
+        };
+        let savedContext = new SpanContext(
+            Utils.encodeInt64(1),
+            Utils.encodeInt64(2),
+            Utils.encodeInt64(3),
+            constants.SAMPLED_MASK,
+            baggage
+        );
+
+        let assertByFormat = (format) => {
+            let carrier = {};
+            tracer.inject(savedContext, format, carrier);
+            let extractedContext = tracer.extract(format, carrier);
+
+            assert.isOk(bufferEqual(savedContext.traceId, extractedContext.traceId));
+            assert.isOk(bufferEqual(savedContext.spanId, extractedContext.spanId));
+            assert.isOk(bufferEqual(savedContext.parentId, extractedContext.parentId));
+            assert.equal(savedContext.flags, extractedContext.flags);
+            assert.equal(savedContext.baggage[keyOne], extractedContext.baggage[keyOne]);
+            assert.equal(savedContext.baggage[keyTwo], extractedContext.baggage[keyTwo]);
+        }
+
+        assertByFormat(opentracing.FORMAT_TEXT_MAP);
+        assertByFormat(opentracing.FORMAT_HTTP_HEADERS);
+    });
+
+    it ('inject url encoded values into headers', () => {
+        let baggage = {
+            keyOne: 'leela',
+            keyTwo: 'bender'
+        };
+        let savedContext = new SpanContext(
+            Utils.encodeInt64(1),
+            Utils.encodeInt64(2),
+            Utils.encodeInt64(3),
+            constants.SAMPLED_MASK,
+            baggage
+        );
+        let carrier = {};
+
+        tracer.inject(savedContext, opentracing.FORMAT_HTTP_HEADERS, carrier);
+        assert.equal(carrier['UberCtx-keyOne'], encodeURIComponent('leela'));
+        assert.equal(carrier['UberCtx-keyTwo'], encodeURIComponent('bender'));
+    });
+
+    it ('assert inject and extract throw errors when given an invalid format', () => {
+        let carrier = {};
+        let context = new SpanContext(
+            Utils.encodeInt64(1),
+            Utils.encodeInt64(2),
+            Utils.encodeInt64(3),
+            constants.SAMPLED_MASK
+        );
+
+        // subtle but expect wants a function to call not the result of a function call.
+        expect(() => {tracer.inject(context, 'fake-format', carrier)}).to.throw('Unsupported format: fake-format');
+        expect(() => {tracer.extract('fake-format', carrier)}).to.throw('Unsupported format: fake-format');
+    });
+});
```

Comment:
missing tests: what happens when malformed trace context string is given to extractor? https://github.com/uber/jaeger-client-go/blob/master/context_test.go#L10